### PR TITLE
[WIN32K] Implement NtUserSetWindowLongPtr for 64 bit builds

### PR DESCRIPTION
**Pull request by tkreuzer**
Also fix return type of NtUserQueryWindow

### CODE REVIEW

File: win32ss/user/ntuser/window.c [hunk 1]
```diff
@@ -3689,39 +3720,39 @@ co_IntSetWindowLong(HWND hWnd, DWORD Index, LONG NewValue, BOOL Ansi, BOOL bAlte
...
             break;
 
-         case GWL_ID:
+         case GWL_ID: // LONG
             OldValue = (LONG) Window->IDMenu;
             Window->IDMenu = (UINT) NewValue;
             break;
 
-         case GWL_USERDATA:
+         case GWL_USERDATA: // LONG
```
Comment (ThFabba) [inline win32ss/user/ntuser/window.c]: This doesn't seem right. Applications often store pointers in here.

Comment (ThFabba) [reply win32ss/user/ntuser/window.c]: Ah, this makes more sense after reading the code above. This is _allowed_ to be called with a LONG. But it must support storing a LONG_PTR, which is what makes this confusing.

File: win32ss/user/ntuser/window.c [hunk 2]
```diff
@@ -3689,39 +3720,39 @@ co_IntSetWindowLong(HWND hWnd, DWORD Index, LONG NewValue, BOOL Ansi, BOOL bAlte
                 co_IntSendMessage(hWnd, WM_STYLECHANGED, GWL_STYLE, (LPARAM) &Style);
             break;
 
-         case GWL_WNDPROC:
+         case GWL_WNDPROC: // LONG_PTR
```
Comment (ThFabba) [inline win32ss/user/ntuser/window.c]: You should switch (at least) the LONG_PTR ones over to `GWLP_xxx`. In PSDK headers, GWL_WNDPROC & friends are not defined for x64.

Comment (ThFabba) [reply win32ss/user/ntuser/window.c]: Many of these need cleaning up too if you're interested ;)
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=GWL_HINSTANCE
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=GWL_WNDPROC
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=GWL_USERDATA

Or we could add this to Jira as a good newbie task.

File: win32ss/w32ksvc.h [hunk 3]
```diff
@@ -544,6 +544,9 @@
     SVC_(UserSetTimer, 4)
     SVC_(UserSetWindowFNID, 2)
     SVC_(UserSetWindowLong, 4)
+#ifdef _WIN64
+    SVC_(UserSetWindowLongPtr, 4)
```
Comment (ThFabba) [inline win32ss/w32ksvc.h]: Is this the right syscall number, or is our syscall order totally broken on x64 anyway? :p

**Comment by HBelusca** [comment]
Is GetWindowLongPtr correct in that regard?

File: sdk/include/psdk/windowsx.h [hunk 4]
```diff
@@ -285,7 +285,7 @@
 #define GetWindowExStyle(hwnd) ((DWORD)GetWindowLong(hwnd,GWL_EXSTYLE))
```
Comment (HBelusca) [inline sdk/include/psdk/windowsx.h]: You are not using GetWindowLongPtr for these ones?

File: sdk/include/psdk/windowsx.h [hunk 5]
```diff
@@ -495,7 +495,7 @@
 #define Static_SetIcon(hwndCtl,hIcon) ((HICON)(UINT)(DWORD)SendMessage((hwndCtl),STM_SETICON,(WPARAM)(HICON)(hIcon),0))
 #define Static_SetText(hwndCtl,lpsz) SetWindowText((hwndCtl),(lpsz))
 #define SubclassDialog(hwndDlg,lpfn) ((DLGPROC)SetWindowLong(hwndDlg,DWL_DLGPROC,(LPARAM)(DLGPROC)(lpfn)))
```
Comment (HBelusca) [inline sdk/include/psdk/windowsx.h]: Shouldn't this one be using SetWindowLongPtr too?

File: win32ss/user/user32/windows/class.c [hunk 6]
```diff
@@ -1841,7 +1841,7 @@ SetWindowLongPtrA(HWND hWnd,
                   INT nIndex,
                   LONG_PTR dwNewLong)
 {
-    return NtUserSetWindowLong(hWnd, nIndex, dwNewLong, FALSE);
+    return NtUserSetWindowLongPtr(hWnd, nIndex, dwNewLong, FALSE);
```
Comment (HBelusca) [inline win32ss/user/user32/windows/class.c]: The last boolean should be TRUE here (calling the function in ANSI mode).
The other call in SetWindowLongPtrW is OK however.

Comment (HBelusca) [inline win32ss/user/user32/windows/class.c]: The last boolean should be TRUE here (calling the function in ANSI mode).
The other call in SetWindowLongPtrW is OK however.

File: sdk/include/psdk/windowsx.h (hunk 4 above)
Comment (tkreuzer) [reply sdk/include/psdk/windowsx.h]: The style is always 32 bits.

File: sdk/include/psdk/windowsx.h (hunk 5 above)
Comment (tkreuzer) [reply sdk/include/psdk/windowsx.h]: Yes, will fix

File: win32ss/user/user32/windows/class.c (hunk 6 above)
Comment (tkreuzer) [reply win32ss/user/user32/windows/class.c]: Will fix

**Pull request merged**